test/unit/deferred.js

```diff
@@ -525,6 +525,34 @@ QUnit.test( "jQuery.Deferred.then - spec compatibility", function( assert ) {
 	} catch ( _ ) {}
 } );
 
+QUnit[ window.console ? "test" : "skip" ]( "jQuery.Deferred.exceptionHook", function( assert ) {
```

Comment:
If you want to come up for a spec for `QUnit.testIf()` we can polyfill it until we upgrade, or we could upgrade QUnit and make the change then. Let's do that change outside this ticket though.